lib/analyze-scope.js

```diff
@@ -308,6 +308,10 @@ class Referencer extends OriginalReferencer {
 }
 
 module.exports = function(ast, parserOptions) {
+  if (OriginalReferencer._babelEslintPatched) {
+    return escope.analyze(ast, parserOptions);
+  }
+
   const options = {
```

Comment:
I think `escope.analyze` cannot understand `parserOptions`. It needs this `options`.

Reply:
I agree, but just passing in that `options` object causes some additional errors. I think this needs to use the same option-translation logic that `parse-with-patch` uses.

Reply:
It seems like the monkeypatching in `parse-with-patch` relies on a global `eslintOptions` object, which isn't being set properly here because the `parse-with-patch` function isn't being called. To fix it, I think it would be better to pass the options down directly.

Reply:
I'm investigating it as well. But I have to leave while an hour.